### PR TITLE
plasma6: mark dolphin + spectacle as optional packages & reorganize dependencies

### PR DESCRIPTION
**Pull request by omentic**
## Description of changes

This moves `dolphin`, `dolphin-plugins`, and `spectacle` into the optional packages section of the plasma6 module. These are not necessary for Plasma's operation, and as they're exposed to the user through desktop entries - people may not want them to be strictly required.

It also reorders some dependencies and adds a few lines of documentation about what they're there for.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by drupol — COMMENTED** [review]
I had a quick glance at this PR and I notice that the scope of it has been extended and you reorganized a little bit things that shouldn't be part of the single commit in here. At work I usually block this kind of PR.

For the reviewers, wouldn't it be better to just create a single commit containing the changes related to the PR (mark things as optionals) and then only another commit that reformat/reorganize things around?

**Comment by omentic** [comment]
Sure thing, rebased :+1: 

There doesn't happen to be any kind of existing system for tying descriptive messages to dependencies, by any chance?

**Review by Shawn8901 — APPROVED** [review]
looks reasonable now to me.

**Review by pluiedev — APPROVED** [review]
Might as well rename the PR to also mention the second commit, but otherwise LGTM

File: nixos/modules/services/desktop-managers/plasma6.nix [hunk 1]
```diff
@@ -87,30 +88,27 @@ in {
...
+        ffmpegthumbs # video thumbnailer
+        kdegraphics-thumbnailers # pdf etc thumbnailer
+        polkit-kde-agent-1 # polkit auth ui
         plasma-desktop
         plasma-workspace
-
-        # Crash handler
-        drkonqi
+        drkonqi # crash handler
+        kde-inotify-survey # handles running out of inodes
```
Comment (K900) [inline nixos/modules/services/desktop-managers/plasma6.nix:106]: That's not what this does

File: nixos/modules/services/desktop-managers/plasma6.nix [hunk 2]
```diff
@@ -87,30 +88,27 @@ in {
...
-        kglobalacceld
+        kglobalacceld # keyboard shortcut daemon
         kwrited # wall message proxy, not to be confused with kwrite
-
-        milou
-        polkit-kde-agent-1
-
+        baloo # system indexer
+        milou # search engine atop baloo
+        ffmpegthumbs # video thumbnailer
```
Comment (K900) [inline nixos/modules/services/desktop-managers/plasma6.nix:100]: Really doesn't make much sense to have this non-optional and Dolphin optional

File: nixos/modules/services/desktop-managers/plasma6.nix [hunk 3]
```diff
@@ -75,6 +73,9 @@ in {
         kiconthemes # provides Qt plugins
         kimageformats # provides Qt plugins
         kio # provides helper service + a bunch of other stuff
+        kio-admin # managing files as admin
+        kio-extras # stuff for MTP, AFC, etc
+        kio-fuse # fuse interaction
```
Comment (K900) [inline nixos/modules/services/desktop-managers/plasma6.nix:78]: Very vague.

File: nixos/modules/services/desktop-managers/plasma6.nix (hunk 1 above)
Comment (omentic) [reply nixos/modules/services/desktop-managers/plasma6.nix:106]: kde-inotify-survey? It warns the user when they run out of inodes, no?

Comment (K900) [reply nixos/modules/services/desktop-managers/plasma6.nix:106]: No, it warns the user about low inotify watch limits.

**Comment by omentic** [comment]
Thanks!

**Pull request merged**